research/conditional/utils/model_utils.py

```diff
@@ -242,6 +246,10 @@ def get_expert_choice_args(args):
         "group_size": args.simulate_group_size,
         "init_type": args.init_type,
         "init_scale": args.init_scale,
+        "fsdp_enabled": args.fsdp,
+        "rank": rank,
+        "param_precision": torch.bfloat16 if args.mixed_precision else torch.float32,
```

Comment:
this seems new. Have we not veen using `torch.float16` up to this point for MP? Has this been consulted with anyone?

Reply:
`bfloat16` is generally better suited for Machine Learning. Our use of `float16` dates back to the time when we did't have access A100. Currently only Ampere Nvidia architecture (A100, H100) supports this `dtype`

Reply:
as discussed below, let's discuss this with the team